### DISCUSSION TITLE
Support for rustls?

### DISCUSSION THREAD

**Discussion by stevefan1999-personal** [tlsfuzzer/tlsfuzzer#894] [General]
I want to see if tlsfuzzer can be used to fuzz [rustls](https://github.com/rustls/rustls), but I don't know how to port it

**Comment by ueno** [comment]
I guess you would need to create a [harness](https://github.com/tlsfuzzer/tlsfuzzer/wiki/Server-setup) (simple TLS server with some configuration options) backed by rustls. One tricky thing I can think of is that quite a few tests assume that the server behaves as an HTTPS server, so maybe you might rather need to look at [hyper-rustls](https://github.com/rustls/hyper-rustls) for example.

**Reply by tomato42** [reply]
While the test do send an HTTP request, they just expect a reply back, they don't actually parse or validate the reply, so a simple echo server will work just as well. (There may be one or two tests that require a real HTTP server, like the one for server-initiated renegotiation based on requested resource, but those are an exception, not the norm).
